bson/src/test/unit/org/bson/types/Decimal128Specification.groovy

```diff
@@ -31,11 +32,11 @@ class Decimal128Specification extends Specification {
 
     def 'should have correct constants'() {
         expect:
-        POSITIVE_ZERO == fromIEEE754BIDEncoding(0x3040000000000000L, 0x0000000000000000L)
-        NEGATIVE_ZERO == fromIEEE754BIDEncoding(0xb040000000000000L, 0x0000000000000000L)
-        POSITIVE_INFINITY == fromIEEE754BIDEncoding(0x7800000000000000L, 0x0000000000000000L)
-        NEGATIVE_INFINITY == fromIEEE754BIDEncoding(0xf800000000000000L, 0x0000000000000000L)
-        NaN == fromIEEE754BIDEncoding(0x7c00000000000000L, 0x0000000000000000L)
+        POSITIVE_ZERO.equals(fromIEEE754BIDEncoding(0x3040000000000000L, 0x0000000000000000L))
```

Comment:
I no longer trust Spock/Groovy to actually call an equals method when using `==`.  I think it will use compareTo when available.

Should I just re-write the whole thing in Java?

Reply:
Too much magic in groovy - I'd be tempted to convert to Java.